src/CMakeLists.txt

```diff
@@ -418,6 +423,91 @@ add_library(Halide
+        FetchContent_Declare(
+            flatbuffers
+            GIT_REPOSITORY https://github.com/google/flatbuffers.git
+            GIT_TAG v${FLATBUFFERS_VER}
+            GIT_SHALLOW TRUE
+        )
+        # configuration for flatbuffers
+        set(FLATBUFFERS_BUILD_TESTS OFF)
+        set(FLATBUFFERS_INSTALL OFF)
+        FetchContent_MakeAvailable(flatbuffers)
+        set_target_properties(flatbuffers PROPERTIES POSITION_INDEPENDENT_CODE ON)
+
+        add_library(Halide_flatbuffers INTERFACE)
+        target_sources(Halide_flatbuffers INTERFACE $<BUILD_INTERFACE:$<TARGET_OBJECTS:flatbuffers>>)
+        target_include_directories(Halide_flatbuffers
+                                   SYSTEM # Use -isystem instead of -I; this is a trick so that clang-tidy won't analyze these includes
+                                   INTERFACE
+                                   $<BUILD_INTERFACE:${flatbuffers_SOURCE_DIR}>/include
+                                   $<BUILD_INTERFACE:${flatbuffers_BINARY_DIR}>/include)
+        set_target_properties(Halide_flatbuffers PROPERTIES EXPORT_NAME flatbuffers)
+
+        add_executable(flatbuffers::flatc ALIAS flatc)
+        message(STATUS "Using fetched-and-built flatbuffers, version ${FLATBUFFERS_VER}")
+    else ()
+        # Sadly, there seem to be at least three variations of the Flatbuffer package
+        # in terms of the case of the relevant CMake files; if we guess wrong, we
+        # fail on case-sensitive file systems. We'll try this as a hack workaround:
+        # just try all three. (Note that the internal CMake library name appears to be
+        # `flatbuffers` in all cases.)
+        set(FB_NAME "")
+        foreach (N IN ITEMS flatbuffers Flatbuffers FlatBuffers)
+            # TODO: should we check the version here?
+            find_package(${N} QUIET)
+            if (${N}_FOUND)
+                set(FB_NAME ${N})
+                message(STATUS "Using installed flatbuffers, version ${${N}_VERSION}")
+                break()
+            endif ()
+        endforeach ()
+
+        if (NOT FB_NAME)
+            message(FATAL_ERROR "WITH_SERIALIZATION is ON and FLATBUFFERS_USE_FETCHCONTENT is OFF, "
+                                "but could not find flatbuffers installed locally. "
+                                "Either install flatbuffers or build with WITH_SERIALIZATION=OFF.")
+        endif ()
+
+        add_library(Halide_flatbuffers ALIAS flatbuffers::flatbuffers)
+    endif ()
+
+    set(fb_dir "${Halide_BINARY_DIR}/flatc/include")
+
+    set(fb_def "${CMAKE_CURRENT_SOURCE_DIR}/halide_ir.fbs")
+    set(fb_header "${fb_dir}/halide_ir_generated.h")
+    add_custom_command(
+        OUTPUT "${fb_header}"
+        COMMAND flatbuffers::flatc --cpp -o "${fb_dir}" "${fb_def}"
+        DEPENDS "${fb_def}"
+        VERBATIM
+    )
+    add_custom_target(generate_fb_header DEPENDS "${fb_header}")
+    set_source_files_properties("${fb_header}" PROPERTIES GENERATED TRUE)
```

Comment:
Why is this necessary? Add custom command sets it automatically

Reply:
I didn't realize that.